src/schoology-extractor/tests/data_generation/__main__.py

```diff
@@ -23,6 +37,79 @@
 )
 logger = logging.getLogger(__name__)
 
-NUMBER_OF_USERS = 60
+request_client: RequestClient = RequestClient(
+    arguments.client_key, arguments.client_secret
+)
+
+
+grading_periods = []
+try:
+    grading_periods = get_gradingperiods(request_client)
+except Exception as ex:
+    traceback.print_exception(type(ex), ex, ex.__traceback__)
+
+
+NUMBER_OF_USERS = 5
+users = []
+try:
+    users = generate_and_load_users(request_client, NUMBER_OF_USERS)
+except Exception as ex:
+    traceback.print_exception(type(ex), ex, ex.__traceback__)
+
+
+NUMBER_OF_COURSES = 5
+courses = []
+try:
+    courses = generate_and_load_courses(request_client, NUMBER_OF_COURSES)
+except Exception as ex:
+    traceback.print_exception(type(ex), ex, ex.__traceback__)
```

Comment:
If the load of courses fails, does it really make sense to continue on and try to create sections?

Reply:
I wasn't really worrying about it too much, since I'd expect dependencies of failures to just fail as well.